pkg/kubernetes/mtls.go

```diff
@@ -107,6 +110,31 @@ func ExportTrustChain(outputDir string) error {
 	return nil
 }
 
+// Check and warn if cert expiry is less than `warningDaysForCertExpiry` days.
+func CheckForCertExpiry() {
+	expiry, err := Expiry()
+	// The intent is to warn for certificate expiry only when it can be fetched.
+	// Do not show any kind of errors with normal command flow.
+	if err == nil {
+		daysRemaining := int(expiry.Sub(time.Now().UTC()).Hours() / 24)
+		if daysRemaining < warningDaysForCertExpiry {
+			warningMessage := ""
+			if daysRemaining == 0 {
+				warningMessage = "Root certificate of your kubernetes cluster expires today"
+			} else if daysRemaining < 0 {
+				warningMessage = "Root certificate your kubernetes cluster already expired"
+			} else {
+				warningMessage = fmt.Sprintf("Root certificate your kubernetes cluster expires in %v days", daysRemaining)
+			}
+			color.Set(color.FgHiYellow)
+			helpMessage := "Kindly renew to avoid any service interuptions."
+			message := fmt.Sprintf("%s. Expiry date: %s. \n %s", warningMessage, expiry.Format(time.RFC1123), helpMessage)
+			print.WarningStatusEvent(os.Stdout, message)
```

Comment:
There are colors defined in the `print` pkg which look unused. As a caller of `WarningStatusEvent`, I would rely on the function to set the colors instead. It can use different colors based on the OS, environment, accessibility guidelines, etc., but the caller should remain agnostic. What are your thoughts, @pravinpushkar, @mukundansundar?

Reply:
the caller should remain agnostic of the colour ... Ideally only the behaviour that is wanted ... The colour will be set internally as defined by the function and needs to be consistent across all such calls.

Reply:
Thanks for sharing your thoughts. For this PR maybe we can omit the color part then, and track the behavior of `print` pkg separately? I see color being used in only one more place that might need refactoring.

https://github.com/dapr/cli/blob/f89be62068c39ffd3492ab24025e3c0624a56c61/pkg/standalone/standalone.go#L788

Reply:
I am also ok to remove the color. Just a thought,  normally people use blackish/whitish background terminal and in both these theme yellow(it is not actually yellow, it is kind of orange) would fit nicely at the same time we do want to highlight this because of the seriousness.

Reply:
It will be better to have diff PR. I will remove the color part.